python/tk_desktop/no_apps_installed_overlay.py

```diff
@@ -94,14 +95,30 @@ def _load_software_icons(self, engine, project=None):
         icon_row_layout = None
         last_software = len(sg_softwares) - 1
         for i in range(len(sg_softwares)):
-            # Download the thumbnail file from Shotgun.
-            sg_icon = shotgun_data.ShotgunDataRetriever.download_thumbnail(
-                sg_softwares[i]["image"], engine
-            )
+            try:
+                # Download the thumbnail source file from Shotgun, which preserves
+                # transparency and alpha values.
+                sg_icon = shotgun_data.ShotgunDataRetriever.download_thumbnail_source(
+                    sg_softwares[i]["type"], sg_softwares[i]["id"], engine
+                )
+
+            except (TankError, AttributeError):
+                # An old version of tk-framework-shotgunutils or tk-core is 
+                # likely in use. Try ShotgunDataRetriever.download_thumbnail() instead.
+                engine.logger.warning(
+                    "ShotgunDataRetriever is unable to download thumbnail source file. "
```

Comment:
Maybe the message could say something about what the end result is, i.e. having icons with no transparency. This would help the user understand why upgrading is important.